src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs

```diff
@@ -1198,5 +1208,454 @@ static void verifySymbolInfo(IMethodSymbol methodSymbol, PublicNullableAnnotatio
+                Assert.Equal(expectedAnnotation, symbol.NullableAnnotation);
+            }
+        }
+
+        [Fact]
+        public void GetDeclaredSymbol_Fixed()
+        {
+            var source = @"
+class C
+{
+    unsafe void M(object? o)
+    {
+        fixed (var o1 = o ?? new object())
+        {
+        }
+    }
+}
+";
+
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue().WithAllowUnsafe(true));
+            comp.VerifyDiagnostics(
+                // (6,20): error CS0821: Implicitly-typed local variables cannot be fixed
+                //         fixed (var o1 = o ?? new object())
+                Diagnostic(ErrorCode.ERR_ImplicitlyTypedLocalCannotBeFixed, "o1 = o ?? new object()").WithLocation(6, 20));
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var root = syntaxTree.GetRoot();
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var declaration = root.DescendantNodes().OfType<VariableDeclaratorSyntax>().Single();
+            var symbol = (ILocalSymbol)model.GetDeclaredSymbol(declaration);
+            Assert.Equal(PublicNullableAnnotation.NotAnnotated, symbol.NullableAnnotation);
+        }
+
+        [Fact]
+        public void GetDeclaredSymbol_ForLoop()
+        {
+            var source = @"
+class C
+{
+    void M(object? o1, object o2)
+    {
+        for (var o3 = o1; false; ) {}
+        for (var o4 = o1 ?? o2; false; ) {}
+        for (var o5 = o1, o6 = o2; false; ) {}
+    }
+}
+";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (8,14): error CS0819: Implicitly-typed variables cannot have multiple declarators
+                //         for (var o5 = o1, o6 = o2; false; ) {}
+                Diagnostic(ErrorCode.ERR_ImplicitlyTypedVariableMultipleDeclarator, "var o5 = o1, o6 = o2").WithLocation(8, 14));
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var root = syntaxTree.GetRoot();
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var declarations = root.DescendantNodes().OfType<VariableDeclaratorSyntax>().ToList();
+
+            assertAnnotation(declarations[0], PublicNullableAnnotation.Annotated);
+            assertAnnotation(declarations[1], PublicNullableAnnotation.NotAnnotated);
+            assertAnnotation(declarations[2], PublicNullableAnnotation.Annotated);
+            assertAnnotation(declarations[3], PublicNullableAnnotation.NotAnnotated);
+
+            void assertAnnotation(VariableDeclaratorSyntax variable, PublicNullableAnnotation expectedAnnotation)
+            {
+                var symbol = (ILocalSymbol)model.GetDeclaredSymbol(variable);
+                Assert.Equal(expectedAnnotation, symbol.NullableAnnotation);
+            }
+        }
+
+        [Fact]
+        public void GetDeclaredSymbol_OutVariable()
```

Comment:
>GetDeclaredSymbol_OutVariable [](start = 20, length = 29)

A variant of this test with type inference would be great:
`void Out<T>(T input, out T output)`